### PR TITLE
feat: Trigger replay on SLO creation

### PR DESCRIPTION
**Pull request by greg-agacinski**
## Motivation
Allow Nobl9 users to automatically launch historical data retrieval (Replay) when creating new SLOs using the Nobl9 Terraform Provider.

## Summary

Introduces the ability to trigger a replay (historical data retrieval) for a newly created SLO by adding the `retrieve_historical_data_from` parameter. Appropriate messaging is provided to the user.

**Caveats:**

- The SLO is created regardless of the success of the Replay trigger.  
- Replay availability for the datasource used by the SLO is not validated.  
- All Replay errors are printed.

**Example**
`retrieve_historical_data_from  = "2024-11-14T13:00:00Z"`

This feature is most useful when aligned with the Playlist (Replay queues) feature; however, it is fully functional with current slot limitations.

Zrzut ekranu 2024-11-15 o 10 57 48

## Testing

Creating and updating multiple SLOs with datasources that both support and don't support replay. Also, try creating more than the available slots.

## Release Notes

Add the ability to trigger historical data retrieval when creating a new SLO.

### CODE REVIEW

**Comment by paulina-n9** [comment]
I've applied and re-applied SLOs with datasources that support replay.

I've also checked the negative scenarios such as:
- no available slots (when playlists are disabled)
Zrzut ekranu 2024-11-21 o 14 05 45

- maxHistoricalDataRetrival too low
Zrzut ekranu 2024-11-21 o 14 18 06

- data surce does not support replay
Zrzut ekranu 2024-11-21 o 14 21 33

**Comment by greg-agacinski** [comment]
@paulina-n9 added https://github.com/nobl9/terraform-provider-nobl9/pull/342/commits/4b10b8e6fb171b7911ca4f4743d8e16600f44efe to print out human readable (pretty) errors.

File: nobl9/resource_slo.go [hunk 1]
```diff
@@ -330,19 +335,30 @@ func schemaSLO() map[string]*schema.Schema {
 			},
 		},
 		"anomaly_config": schemaAnomalyConfig(),
+		"retrieve_historical_data_from": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validateDateTime,
+			Description: "If set, the retrieval of historical data for a newly created SLO will be triggered, " +
+				"starting from the specified date.",
```
Comment (kskitek) [inline nobl9/resource_slo.go:343]:
```suggestion
			Description: "If set, the retrieval of historical data for a newly created SLO will be triggered, " +
				"starting from the specified date. Needs to be RFC3339 format.",
```

**Review by kskitek — APPROVED** [review]
Looks good, I have tested it very briefly so it might require a full round of tests.

**Pull request merged**